### PR TITLE
Raspbian Backend (1st phase)

### PR DESCRIPTION
**Pull request by ritwickdsouza**
Merging raspbian branch of fork to rasbian branch of original repo.

References #78.

### CODE REVIEW

File: netjsonconfig/backends/raspbian/converters/base.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from ...base.converter import BaseConverter
+
+
+class RaspbianConverter(BaseConverter):
+    def test_function(self):
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/converters/base.py]: what's this for?

File: netjsonconfig/backends/raspbian/raspbian.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from .renderer import Commands, Hostapd, Hostname, Interfaces, Ntp, Resolv
+from .schema import schema
+
+
+class Raspbian(BaseBackend):
+    """
+    Raspbian Backend
+    """
+    schema = schema
+    env_path = 'netjsonconfig.backends.raspbian'
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/raspbian.py]: this shouldn't be needed anymore, try removing it

File: netjsonconfig/backends/raspbian/templates/hostname.jinja2 [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% for i, j in data.items() %}
+{% if i|string() == 'general' %}
+{% for general in j %}
+{% if general.get('hostname') %}
+config: /etc/hostname
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/templates/hostname.jinja2]: are you sure this line doesn't make the configuration invalid when deployed on an actual device?

Shouldn't it be `# config: /etc/hostname`?

File: netjsonconfig/backends/raspbian/templates/interfaces.jinja2 [hunk 4]
```diff
@@ -0,0 +1,83 @@
+{% for i, j in data.items() %}
+{% if i|string() == 'interfaces' %}
+{% for interface in j %}
+{% if interface.get('iftype') == 'wireless' %}
+{% if interface.get('mode') == 'adhoc' %}
+config: /etc/network/interfaces
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/templates/interfaces.jinja2]: are you sure this line doesn't make the configuration invalid when deployed on an actual device?

Shouldn't it be `# config: /etc/network/interfaces`?

File: netjsonconfig/backends/raspbian/templates/ntp.jinja2 [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{% for i, j in data.items() %}
+{% if i|string() == 'ntp' %}
+config: /etc/ntp.conf
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/templates/ntp.jinja2]: same as previous comment

File: netjsonconfig/backends/raspbian/templates/resolv.jinja2 [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{% set count = [1] %}
+{% for i, j in data.items() %}
+{% if i|string() in ['dns_servers', 'dns_search'] %}
+{% if count == [1] %}
+config: /etc/resolv.conf
```
Comment (nemesisdesign) [inline netjsonconfig/backends/raspbian/templates/resolv.jinja2]: same as previous comment

File: tests/raspbian/test_hostapd.py [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+                        }
+                    }
+                }
+            ]
+        })
+
+        expected = ''''''
+        self.assertEqual(o.render(), expected)
+
+    @unittest.skip('Test skipping')
```
Comment (nemesisdesign) [inline tests/raspbian/test_hostapd.py]: what about this test skipping thing?

File: tests/raspbian/test_hostapd.py [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+driver=nl80211
+hw_mode=g
+channel=3
+ieee80211n=1
+ssid=open
+
+'''
+        self.assertEqual(o.render(), expected)
+
+    @unittest.skip('Test skipping')
```
Comment (nemesisdesign) [inline tests/raspbian/test_hostapd.py]: same as previous comment

File: tests/raspbian/test_hostapd.py [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+ssid=wpa-personal
+auth_algs=1
+wpa=1
+wpa_key_mgmt=WPA-PSK
+wpa_passphrase=passphrase012345
+
+'''
+        self.assertEqual(o.render(), expected)
+
+    @unittest.skip('Test skipping')
```
Comment (nemesisdesign) [inline tests/raspbian/test_hostapd.py]: what about this test skipping thing?

Comment (ritwickdsouza) [inline tests/raspbian/test_hostapd.py]: These are not implemented as of now. So skipping them till I implement it.

File: tests/raspbian/test_hostapd.py (hunk 8 above)
Comment (ritwickdsouza) [inline tests/raspbian/test_hostapd.py]: Not implemented

File: tests/raspbian/test_hostapd.py (hunk 7 above)
Comment (ritwickdsouza) [inline tests/raspbian/test_hostapd.py]: Not implemented

File: netjsonconfig/backends/raspbian/templates/hostname.jinja2 (hunk 3 above)
Comment (ritwickdsouza) [inline netjsonconfig/backends/raspbian/templates/hostname.jinja2]: Yes. I'll comment it out.

**Pull request merged**